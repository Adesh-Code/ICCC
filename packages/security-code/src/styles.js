const styles = ({ tokens, primitives }) => ({
  "&.DhigSecurityCode": {
    width: "56px",
    height: "72px",
    borderTop: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`,
    borderBottom: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"][class*="MuiInputBase-inputAdornedEnd"]':
    {
      marginRight: "0"
    },
  "&.DhigSecurityCode[class*='MuiInput-root']:hover:before": {
    height: "2px",
    backgroundColor: tokens.colors["fill-engage"],
    boxShadow: "none",
    borderBottom: `${primitives.borderWidths.medium} solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode:hover": {
    boxShadow: "none"
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.primary.white}`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode input:focus": {
    boxShadow: `0px 2px 0px 0px ${tokens.colors["ink-focus"]} inset, 0px 0px 0px 2px ${tokens.colors["ink-focus"]} inset`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:not(.Mui-disabled)::before': {
    boxShadow: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]:before': {
    width: "56px"
  },
  "&.DhigSecurityCode input": {
    flexGrow: "1"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "56px",
    height: "72px",
    fontSize: `${primitives.fontSizeFallbacks["6"]}`,
    fontWeight: primitives.fontWeights.default
  },
  '&.DhigSecurityCode--dense [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "32px",
    height: "40px",
    fontSize: `${primitives.fontSizeFallbacks["3"]}`,
    fontWeight: primitives.fontWeights.default
  },
  "&.DhigSecurityCode--dense": {
    width: "32px",
    height: "40px"
  },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode--error, &.DhigSecurityCode:first-of-type.DhigSecurityCode--error, &.DhigSecurityCode:last-child.DhigSecurityCode--error, &.DhigSecurityCode--error:before":
    {
      borderColor: primitives.colors.accent.clay
    },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:after': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode:first-of-type": {
    borderLeft: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child": {
    borderRight: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child [class*='MuiDivider-root'], &.DhigSecurityCode:focus-within [class*='MuiDivider-root']":
    {
      display: "none"
    }
});

export default styles;
