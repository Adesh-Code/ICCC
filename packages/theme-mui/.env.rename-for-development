# To load the MUI X Pro license for local development, rename this file to .env and replace with the actual license string
DHIG_MUIXPRO_ENTRY=PASTE_MUI_X_PRO_LICENSE_STRING_HERE