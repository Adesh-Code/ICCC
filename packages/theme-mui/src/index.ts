// Themes
export { default as themeDark } from "./themes/dark";
export { default as themeLight } from "./themes/light";

// Utilities
export { configureClassnameGenerator } from "./classname-generator";
export { setLicenseKey } from "./set-license-key";
