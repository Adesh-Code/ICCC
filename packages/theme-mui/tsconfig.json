{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "lib/esm",
    "resolveJsonModule": true,
    "declarationDir": "lib/types"
  },
  "include": ["src"]
}
