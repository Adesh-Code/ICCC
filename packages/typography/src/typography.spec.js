// Libraries
import React from "react";

// Components
import Typography from "./typography";
import { themeLight, themeDark } from "@iccc/theme-mui";
import { createTheme, ThemeProvider } from "@mui/material/styles";

// Test Utilities
import { render, screen } from "@testing-library/react";
import { matchers } from "@emotion/jest";

expect.extend(matchers);

const light = createTheme(themeLight);
const dark = createTheme(themeDark);

const arrange = (options) => {
  const args = {
    children: "Lorem ipsum",
    ...options
  };

  return render(
    <ThemeProvider theme={light}>
      <Typography {...args} />
    </ThemeProvider>
  );
};

describe("Typography component", () => {
  beforeEach(() => {
    // eslint-disable-next-line no-console
    console.error = jest.fn((error) => {
      throw new Error(error);
    });
  });

  it("exposes internal paragraph DOM node", () => {
    // Arrange
    const ref = React.createRef();
    arrange({ ref });

    // Assert
    expect(ref.current).toBeInstanceOf(HTMLParagraphElement);
  });

  it("renders with body-copy-medium variant by default", () => {
    arrange();
    const typography = screen.getByText("Lorem ipsum");
    expect(typography).toHaveStyleRule(
      "font-family",
      "ArtifaktElement,Arial,sans-serif"
    );
    expect(typography).toHaveStyleRule("font-size", /(1rem)/i);
    expect(typography).toHaveStyleRule("font-weight", "400");
    expect(typography).toHaveStyleRule("line-height", "1.5");
    expect(typography).toMatchSnapshot();
  });

  it("renders with different variants", () => {
    arrange({
      variant: "display-extra"
    });
    const typography = screen.getByText("Lorem ipsum");
    expect(typography).toHaveStyleRule("font-family", "ArtifaktLegend");
    expect(typography).toHaveStyleRule("font-size", /(clamp)/i);
    expect(typography).toHaveStyleRule("font-weight", "800");
    expect(typography).toHaveStyleRule("line-height", "1.2");
    expect(typography).toMatchSnapshot();
  });

  it("accepts classnames", () => {
    arrange({
      className: "bar"
    });
    const typography = screen.getByText("Lorem ipsum");
    expect(typography).toHaveClass("bar");
  });

  it("renders with tabular numeric", () => {
    arrange({
      tabularNumeric: true
    });
    const typography = screen.getByText("Lorem ipsum");
    expect(typography).toHaveStyleRule("font-variant-numeric", "tabular-nums");
  });

  it("uses the color prop to handle color", () => {
    arrange({
      color: "ink-error"
    });
    const typography = screen.getByText("Lorem ipsum");
    expect(typography).toHaveStyle("color: rgb(195, 73, 34)");
  });

  it("handles color correctly in dark theme", () => {
    render(
      <ThemeProvider theme={dark}>
        <Typography color="ink-error">Lorem ipsum</Typography>
      </ThemeProvider>
    );

    const typography = screen.getByText("Lorem ipsum");
    expect(typography).toHaveStyle("color: rgb(217, 85, 41)");
  });
});
