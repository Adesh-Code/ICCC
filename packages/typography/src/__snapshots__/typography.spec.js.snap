// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography component renders with body-copy-medium variant by default 1`] = `
<p
  class="MuiTypography-root MuiTypography-body-copy-medium emotion-0"
>
  Lorem ipsum
</p>
`;

exports[`Typography component renders with different variants 1`] = `
<p
  class="MuiTypography-root MuiTypography-display-extra emotion-0"
>
  Lorem ipsum
</p>
`;
